File: ALGORITHMS/JAVA/appleorange.java
public class appleorange{
    static void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges) {
        int in_apple = 0;
        int in_orange = 0;
        for (int i = 0; i < apples.length; i++){
            if ( ((apples[i] + a) <= t) && ((apples[i] + a) >= s) ){
                ++in_apple;
            }
        }

        for (int i = 0; i < oranges.length; i++){
            if ( ((oranges[i] + b) <= t) && ((oranges[i] + b) >= s) ){
                ++in_orange;
            }
        }

        System.out.println(in_apple);
        System.out.println(in_orange);



    }
    public static void main (String args[]){
        int s = 7;
        int t = 11;
        int a = 5;
        int b = 15;
        int [] apples = {-2, 2, 1};
        int [] oranges = {5, -6};

        countApplesAndOranges(s, t, a, b, apples, oranges);
    }
}

File: Extra/PYTHON/appleorange.py
# Complete the countApplesAndOranges function below.
def countApplesAndOranges(s, t, a, b, apples, oranges):
    in_orange = 0
    in_apple = 0
    for i in range(len(apples)):
        if (apples[i] + a) <= t and (apples[i] + a) >= s:
            in_apple = in_apple + 1

    for i in range(len(oranges)):
        if (oranges[i] + b) <= t and (oranges[i] + b) >= s:
            in_orange = in_orange + 1
    
    print (in_apple) 
    return in_orange



#! s: integer, starting point of Sam's house location.
#! t: integer, ending location of Sam's house location.
#! a: integer, location of the Apple tree.
#! b: integer, location of the Orange tree.
#! apples: integer array, distances at which each apple falls from the tree.
#! oranges: integer array, distances at which each orange falls from the tree.


s = 7
t = 10
a = 4
b = 12
apples = [2, 3, -4]
oranges = [3, -2, -4]
print (countApplesAndOranges(s, t , a , b, apples, oranges))